catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/ui/fragment/AddUserbrickTest.java

```diff
@@ -0,0 +1,107 @@
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.catroid.uiespresso.ui.fragment;
+
+import android.support.test.InstrumentationRegistry;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.ViewParent;
+
+import org.catrobat.catroid.ProjectManager;
+import org.catrobat.catroid.R;
+import org.catrobat.catroid.content.Project;
+import org.catrobat.catroid.content.SingleSprite;
+import org.catrobat.catroid.content.Sprite;
+import org.catrobat.catroid.ui.SpriteActivity;
+import org.catrobat.catroid.uiespresso.util.rules.FragmentActivityTestRule;
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.hamcrest.TypeSafeMatcher;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.io.IOException;
+
+import static android.support.test.espresso.Espresso.onData;
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.anything;
+
+public class AddUserbrickTest {
+
+	@Rule
+	public FragmentActivityTestRule<SpriteActivity> baseActivityTestRule = new
+			FragmentActivityTestRule<>(SpriteActivity.class, SpriteActivity.EXTRA_FRAGMENT_POSITION,
+			SpriteActivity.FRAGMENT_SCRIPTS);
+
+	@Before
+	public void setUp() throws IOException {
+		createProject(AddUserbrickTest.class.getName());
+		baseActivityTestRule.launchActivity();
+	}
+
+	private void createProject(String projectName) {
+
+		Project project = new Project(InstrumentationRegistry.getTargetContext(), projectName);
+		Sprite firstSprite = new SingleSprite("first Sprite");
+		project.getDefaultScene().addSprite(firstSprite);
+		ProjectManager.getInstance().setCurrentProject(project);
+		ProjectManager.getInstance().setCurrentSprite(firstSprite);
+	}
+
+	@Test
+	public void checkPlusButtonExists() {
+		onView(withId(R.id.button_add))
+				.perform(click());
+		onData(anything())
```

Comment:
theres already tests accessing the categories, there should be no additional matchers required.

		onData(allOf(is(instanceOf(String.class)), is(UiTestUtils.getResourcesString(R.string.category_motion))))
				.inAdapterView(BrickCategoryListMatchers.isBrickCategoryView())
				.perform(click());

when using on data, use the onData(HERE) to specify the type of data you want to have, and use inAdapterView to only specify the adapter you want to get.

Reply:
Ok, I already checked other tests but I must have overlooked it.
Big thanks for the specific feedback!